core/src/components/datetime/datetime.tsx

```diff
@@ -500,7 +500,16 @@ export class Datetime implements ComponentInterface {
     if (activeParts !== undefined || !isCalendarPicker) {
       const activePartsIsArray = Array.isArray(activeParts);
       if (activePartsIsArray && activeParts.length === 0) {
```

Comment:
Your change should be in the `else` block of this check; currently your code only runs if `multiple="true"`, which isn't compatible with `preferWheel`.

... At least, in theory; I'm honestly not sure why the new test passes here but fails with your change removed 🤨 I did log `activeParts` and `activePartsIsArray` with a `preferWheel` datetime and got an object and `false` locally, so...

Reply:
For some reason if a value is not set on the datetime and using `preferWheel`, the `activeParts` will default to `[]`, which is why this logic works. 

I'll look into the original implementation to identify where this behavior came from and fix it.

Reply:
Yeah I go back on my comment. We shouldn't be relying on the shape of the `activeParts` data to determine if you are in multiple or not. That should be an explicit check of `multiple`. 

I think the original implementation (with refactoring the usage of deconstructing from `this`) is the correct path. Let me know what you think.

Reply:
Ah, that makes sense. In that case, could we add a comment noting that `activeParts` will be `[]` if no value is set, so other people don't get confused for the same reason I did?